File: app/src/main/java/com/parabola/newtone/mvp/view/TabAlbumView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Album;
import com.parabola.domain.settings.ViewSettingsInteractor.AlbumViewType;
import com.parabola.newtone.mvp.CustomAddToEndSingleStrategy;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.StateStrategyType;

@StateStrategyType(CustomAddToEndSingleStrategy.class)
public interface TabAlbumView extends MvpView {
    void refreshAlbums(List<Album> albums);
    void setSectionShowing(boolean enable);

    void setViewType(AlbumViewType viewType);
}

File: app/src/main/java/com/parabola/newtone/presentation/main/albums/TabAlbumView.kt
package com.parabola.newtone.presentation.main.albums

import com.parabola.domain.model.Album
import com.parabola.domain.settings.ViewSettingsInteractor.AlbumItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface TabAlbumView : MvpView {
    fun refreshAlbums(albums: List<Album>)

    fun setAlbumViewSettings(viewSettings: AlbumItemView)
    fun setItemDividerShowing(showed: Boolean)

    fun setSectionShowing(enable: Boolean)
}
